node/runtime/src/impls.rs

```diff
@@ -49,10 +49,18 @@ impl Convert<u128, Balance> for CurrencyToVoteHandler {
 	fn convert(x: u128) -> Balance { x * Self::factor() }
 }
 
-/// Simply multiply by a coefficient denoted by the `Get` implementation.
-impl Convert<Weight, Balance> for WeightToFee {
-	fn convert(x: Weight) -> Balance {
-		Balance::from(x).saturating_mul(Self::get())
+/// Convert from weight to balance via a simple coefficient multiplication
+/// The associated type M encapsulates a constant in units of balance per weight
+pub struct LinearWeight<M>(M);
```

Comment:
```suggestion
pub struct LinearWeightToFee<C>(C);
```

(Sorry if I am taking back my own word :D, if it is a coefficient then C makes more sense. sry)